portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -72,31 +61,58 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
-  // Convert List to JSON using Gson library.
-  private String convertToJson(List<Comment> msgs) {
-    Gson gson = new Gson();
-    String json = gson.toJson(msgs);
-    return json;
-  }
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String newComment = getComment(request);
-    if (newComment != null) {
+    String newCommentText = getComment(request);
+
+    if (newCommentText != null) {
         Entity commentEntity = new Entity("Comment");
-        commentEntity.setProperty("text", newComment);
-        commentEntity.setProperty("time", System.currentTimeMillis());
+        commentEntity.setProperty("text", newCommentText);
+        commentEntity.setProperty("timestamp", getTimestamp());
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         datastore.put(commentEntity);
     }
 
     response.sendRedirect("/index.html");
   }
+
+  // Convert List to JSON using Gson library.
+  private String convertToJson(List<Comment> msgs) {
+    Gson gson = new Gson();
+    String json = gson.toJson(msgs);
+    return json;
+  }
  
  // Extracts comment text from request and returns it.
   private String getComment(HttpServletRequest request) {
     String newComment = request.getParameter("comment");
-    // Add handling for newComment = null?
+
+    // Prevent accidental/blank submissions.
+    if (newComment.equals("")) {
+        return null;
+    }
     return newComment;
   }
+
+// Extracts user-defined limit for number of comments to display.
+  private int getNumberOfCommentsToDisplay(HttpServletRequest request) {
+    String requestValue = request.getParameter("commentLimit");
```

Comment:
I would check here for a missing or empty commentLimit value. For cases where you parse it from the URL text, it might not exist, and thus the value sent will not be a valid integer.

So if you validate it here, you can do the handling instead of trying to parse it and handling it as a failure.

The principle is to try to handle expected inputs directly and save Exceptions for cases when something unexpected happens.

Reply:
Ok, added check when commentLimit is null or an empty value that sets it to the total number of comments.